lua/bufferline/icons.lua

```diff
@@ -36,11 +37,10 @@ return {
     end
  end,
 
-  --- @param buffer_name string
-  --- @param filetype string
+  --- @param bufnr integer
   --- @param buffer_status "Current"|"Inactive"|"Visible"
   --- @return string icon, string highlight_group
-  get_icon = function(buffer_name, filetype, buffer_status)
+  get_icon = function(bufnr, buffer_status)
```

Comment:
The `buffer_name` and `filetype` can be discerned from the `bufnr`, and usually that's what happened at the callsite of this function. Now we just do it automatically.

This was a necessary change because sometimes the `utils.basename(buffer_name)` was being passed in, which would be stripped of the file extension when the `g:bufferline.hide.extensions` setting was enabled.